src/main/cpp/src/config/variant_query_config.cc

```diff
@@ -258,15 +186,138 @@ void VariantQueryConfig::validate(const int rank) {
   if (!m_scan_whole_array && m_row_ranges.size()) {
     VERIFY_OR_THROW((m_single_query_row_ranges_vector || static_cast<size_t>(rank) < m_row_ranges.size())
                     && "Rank >= query row ranges vector size");
-    std::vector<int64_t> row_idxs;
-    for (const auto& range : get_query_row_ranges(rank)) {
-      auto j = row_idxs.size();
-      row_idxs.resize(row_idxs.size() + (range.second - range.first + 1ll));
-      for (auto i=range.first; i<=range.second; ++i,++j)
-        row_idxs[j] = i;
-    }
-    set_rows_to_query(row_idxs);
+    set_rows_to_query(get_query_row_ranges(rank));
   }
   //Attributes
   set_attributes_to_query(m_attributes);
 }
+
+void interval_expander::update_rows(const std::vector<std::pair<int64_t, int64_t>>& vec) {
+  std::vector<interval> temp_intervals;
+  std::swap(temp_intervals, intervals);
+
+  for(auto& e : vec) {
+    int64_t first = e.first, second = e.second;
+    // make sure interval isn't backwards
+    if(e.first <= e.second) {
+      first = e.first;
+      second = e.second;
+    }
+    else {
+      first = e.second;
+      second = e.first;
+    }
+
+    // make sure interval is in bounds (on the left)
+    if(first < 0) {
```

Comment:
If `vec` can be forced into type `std::pair<uint64_t, uint64_t>`, you can avoid this check with first.

Reply:
Kept vec as int64_t to be consistent with TileDBRowRange, but removed check to allow negative values.